lib/Map/UserDrawing.js

```diff
@@ -0,0 +1,400 @@
+     * @type {Bool}
+     * @default true
+     */
+    this.allowPolygon = defaultValue(options.allowPolygon, true);
+
+    /**
+     * Callback that occurs when point is clicked (may be added or removed). Function takes a CustomDataSource which is
+     * a list of PointEntities.
+     * @type {function}
+     * @default undefined
+     */
+    this.onPointClicked = options.onPointClickedCallback;
+
+    /**
+     * Callback that occurs on clean up, i.e. when drawing is done or cancelled.
+     * @type {function}
+     * @default undefined
+     */
+    this.onCleanUp = options.onCleanUpCallback;
+
+    /**
+     * Callback that occurs when the dialog is redrawn, to add additional information to dialog.
+     * @type {function}
+     * @default undefined
+     */
+    this.onMakeDialogMessage = options.onMakeDialogMessageCallback;
+
+    /**
+     * Instance of Terria
+     * @type {Terria}
+     * @default undefined
+     */
+    this.terria = terria;
+
+    /**
+     * Storage for points that will be drawn
+     * @type {CustomDataSource}
+     */
+    this.pointEntities = new CustomDataSource('Points');
+
+    /**
+     * Storage for line that connects the points, and polygon if the first and last point are the same
+     * @type {CustomDataSource}
+     */
+    this.otherEntities = new CustomDataSource('Lines and polygons');
+
+    /**
+     * Polygon that will be drawn if the user drawing is a closed shape
+     * @type {Entity}
+     */
+    this.polygon = undefined;
+
+    /**
+     * Whether to interpret user clicks as drawing
+     * @type {Bool}
+     */
+    this.inDrawMode = false;
+
+    /**
+     * Whether the first and last point in the user drawing are the same
+     * @type {Bool}
+     */
+    this.closeLoop = false;
+
+    /**
+     * SVG element for point drawn when user clicks.
+     * http://stackoverflow.com/questions/24869733/how-to-draw-custom-dynamic-billboards-in-cesium-js
+     */
+    var svgDataDeclare = "data:image/svg+xml,";
+    var svgPrefix = '<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="20px" height="20px" xml:space="preserve">';
+    var svgCircle = '<circle cx="10" cy="10" r="5" stroke="rgb(0,170,215)" stroke-width="4" fill="white" /> ';
+    var svgSuffix = "</svg>";
+    var svgString = svgPrefix + svgCircle + svgSuffix;
+
+    // create the cesium entity
+    this.svgPoint = svgDataDeclare + svgString;
+};
+
+/**
+ * Start interpreting user clicks as placing or removing points.
+ */
+UserDrawing.prototype.enterDrawMode = function() {
+
+    // If we have finished a polygon, don't allow more points to be drawn. In future, perhaps support multiple polygons.
+    if (this.inDrawMode || this.closeLoop) {
+        // Do nothing
+        return;
+    }
+
+    this.inDrawMode = true;
+    document.getElementById("cesiumContainer").setAttribute("style", "cursor: crosshair");
```

Comment:
Don't hardcode `cesiumContainer`.  You can get it from `terria.cesium._mapContainer`.  Might be nice to expose that as a public property instead of accessing the underscore name.